endpoints/openrtb2/auction.go

```diff
@@ -275,6 +275,22 @@ func (deps *endpointDeps) validateImp(imp *openrtb.Imp, aliases map[string]strin
 	return nil
 }
 
+func validateMetric(metrics []openrtb.Metric, impIndex int) error {
+	if len(metrics) == 0 {
+		return nil
+	}
+
+	for i, metric := range metrics {
+		if len(metric.Type) == 0 {
```

Comment:
According to the OpenRTB spec:

> Type of metric being presented using exchange curated string names which should be published to bidders _a priori_

So... these should be a global (and documented) list too.

We should also return an error if the publisher's request uses a Type or Vendor that we don't recognize.